BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -163,65 +169,103 @@ extension SBAProfileItem {
         
         switch self.itemType {
         case SBAProfileTypeIdentifier.string:
-            return newValue as? String != nil
+            return true // anything can be cast to a string
             
         case SBAProfileTypeIdentifier.number:
-            return newValue as? NSNumber != nil
+            if (newValue as? NSNumber != nil) {
+                return true
+            }
+            else if let quantity = newValue as? HKQuantity {
+                return quantity.is(compatibleWith: self.unit ?? commonDefaultUnit())
+            }
+            return false
             
         case SBAProfileTypeIdentifier.bool:
             return newValue as? NSNumber != nil
             
         case SBAProfileTypeIdentifier.date:
-            return newValue as? NSDate  != nil
+            return newValue as? NSDate != nil
             
         case SBAProfileTypeIdentifier.hkBiologicalSex:
             return newValue as? HKBiologicalSex != nil
             
         case SBAProfileTypeIdentifier.hkQuantity:
-            return newValue as? NSNumber != nil
+            guard let quantity = newValue as? HKQuantity else { return false }
+            return quantity.is(compatibleWith: self.unit ?? commonDefaultUnit())
             
         default:
-            return false
+            return true   // Any extended type isn't included in the common validation
```

Comment:
Good point, but this way any value is accepted for any extended type without any kind of validation. Currently we have to add support for any new types here, and then they're supported by all SBAProfileItem implementations. To be honest, I couldn't think of a clean way to extend SBAProfileItem to support new types outside of BridgeAppSDK. The best I could come up with was, if we want to allow that, for each commonXxx() func here we'd make an open xxx() func in SBAProfileItemBase that calls the corresponding commonXxx() and have the classes use those methods instead of the commonXxx() ones directly. Then if you wanted to support an extended type without adding it here, you'd have to subclass whatever profile item class(es) need to support it and override those methods, in which case returning false from super.checkTypeCompatible() would let you know it wasn't a type supported by the parent class so you should do your own checking etc.; and then you'd use your subclasses wherever you needed to support the extended types. If this returns true for unknown types then this approach doesn't work; on the other hand it didn't feel like a particularly elegant or clean approach so I didn't go ahead and build it that way from the start. I couldn't come up with a way to do it via further extensions. Maybe you have a better idea. Anyway I'm OK with this change if we need it for something we're doing to work, just wanted to clarify why I did it the other way originally.